### PR TITLE
Drop old storage engine from code base

### PR DESCRIPTION
**Pull request by ms705**
This pull request removes the old (incomplete and never fully working) in-memory storage engine from the code base. We don't need it any more, and will replace it with different plugins for different use cases:
1. A HDFS adapter for managing references to objects stored in HDFS.
2. A local FS adapter for objects in local directories.
3. At a later stage, a DIOS adapter for DIOS objects accessed via DIOS names and references.

There is no functional impact from this change; I'm merely dropping dead code (and am fixing up a few nasty include order dependencies along the way).
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ms705/firmament/pull/16%23issuecomment-103919542%22%2C%20%22https%3A//github.com/ms705/firmament/pull/16%23issuecomment-104020250%22%2C%20%22https%3A//github.com/ms705/firmament/pull/16%23discussion_r30743130%22%2C%20%22https%3A//github.com/ms705/firmament/pull/16%23discussion_r30757974%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ms705/firmament/pull/16%23issuecomment-103919542%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40ICGog%20Please%20review.%22%2C%20%22created_at%22%3A%20%222015-05-20T15%3A02%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/192315%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ms705%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-05-20T20%3A07%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/433340%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ICGog%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20fb742145646c7410ff12e5d8e7713b5905fcb291%20src/engine/task_lib.cc%2015%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ms705/firmament/pull/16%23discussion_r30743130%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Maybe%20check%20if%20fopen%20failed%20or%20not.%20%28This%20is%20also%20a%20test%20comment%3A%29%20%29.%22%2C%20%22created_at%22%3A%20%222015-05-20T20%3A06%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/433340%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ICGog%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-05-20T22%3A52%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/192315%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ms705%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/engine/task_lib.cc%3AL76-92%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/ICGog%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/433340%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/ICGog'></a>

- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/ms705/firmament/pull/16#issuecomment-103919542'>General Comment</a></b>
- <a href='https://github.com/ms705'></a> @ICGog Please review.
- [x] <a href='#crh-comment-Pull fb742145646c7410ff12e5d8e7713b5905fcb291 src/engine/task_lib.cc 15'></a> &nbsp;<b><a href='https://github.com/ms705/firmament/pull/16#discussion_r30743130'>File: src/engine/task_lib.cc:L76-92</a></b>
- <a href='https://github.com/ICGog'></a> Maybe check if fopen failed or not. (This is also a test comment:) ).

<a href='https://www.codereviewhub.com/ms705/firmament/pull/16?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ms705/firmament/pull/16?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ms705/firmament/pull/16?approve=0'></a>&nbsp;<a href='https://github.com/ms705/firmament/pull/16'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ms705** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ms705/firmament/pull/16'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by ms705** [comment]
@ICGog Please review.

File: src/engine/task_lib.cc [hunk 1]
```diff
@@ -76,16 +76,16 @@ TaskLib::TaskLib()
...
   stringstream ss;
   ss << "/tmp/" << task_id_env << ".pid";
   string pid_filename = ss.str();
 
-  ofstream pid_file;
-  pid_file.open(pid_filename);
-  pid_file << getpid();
-  pid_file.close();
+  FILE* pid_file;
+  pid_file = fopen(pid_filename.c_str(), "w");
```
Comment (ICGog) [inline src/engine/task_lib.cc]: Maybe check if fopen failed or not. (This is also a test comment:) ).

**Comment by ICGog** [comment]
:shipit:

File: src/engine/task_lib.cc (hunk 1 above)
Comment (ms705) [inline src/engine/task_lib.cc]: Addressed in f3883b6b053c96a4be913a3ddf132036322404b4.

Comment (ms705) [inline src/engine/task_lib.cc]: :+1:

**Pull request merged**